src/views/activity/activity_item_view.js

```diff
@@ -21,5 +27,35 @@ export default BaseView.extend({
 
   sendSelectionDetails (event) {
     webChannel.sendUrlSelected(this.model.url, 'url');
+  },
+
+  injectFavicon () {
+    // Manually construct the favicon element and inject it into the DOM,
+    // allowing us to handle events directly, since load events don't bubble
+    // and cannot be delegated.
+    this.icon = new Image('img');
+    this.icon.onload = this.calculateIconColor();
+    this.icon.onerror = this.hideErroredFavicon();
+    this.icon.src = this.model.displayImage();
+    this.iconContainer.appendChild(this.icon);
+  },
+
+  hideErroredFavicon () {
+    return () => {
+      this.icon.style.display = 'none';
+    }.bind(this);
+  },
+
+  calculateIconColor () {
+    // If able to do so in 20ms or less, calculate the dominant color of the
+    // favicon and set that to the background color of its parent.
+    return () => {
+      const START = new Date().getTime();
```

Comment:
Let's swap these out for `performance.now()`, to get in the habit of using high-res timestamps for performance analysis. See [mdn](https://developer.mozilla.org/en-US/docs/Web/API/Performance/now) for details :-)